packages/story-editor/src/components/header/buttons/undo.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+  Icons,
+} from '@web-stories-wp/design-system';
+import { useCallback } from '@web-stories-wp/react';
+import { __ } from '@web-stories-wp/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Tooltip from '../../tooltip';
+import { useConfig, useHistory } from '../../../app';
+
+function Undo() {
```

Comment:
These two buttons seem awfully similar. Could they easily be combined Maybe `<HistoryButton isUndo={false} />`.

Reply:
Hm, since these buttons are not used in the `story-editor` directly at all and are currently meant to be used by external packages using the `story-editor` (such as the WP package), then I actually purposefully split it up so that the external packages could just use it without having to know the internal life of the buttons / history.

We could probably still create a `HistoryButton` as a base and expose `Undo` and `Redo` to the outside though.

Reply:
Changed in e6f9afc